### ISSUE TITLE
Tower Repair

### ISSUE THREAD

**Issue by 89gsc** [kaaku/screeps#3]
I added this in my fork: 

https://github.com/gsc89/screeps/blob/master/src/controller.tower.js

Could you take a look and see if you can spot any issues I currently have it running in one of my rooms and its repairing targets closest to the tower.

I don't think there are issues I just wanted to let you know in case you wish to add this to your own.

**Comment by kaaku** [comment]
I took a look at the code, and I think it looks good. The idea is very similar to what I had in mind. I didn't take it into use as is, however, but I implemented my own version with some small(ish) differences:
- The tower acquires the repair tasks from the task manager instead of doing finds of its own. This helps to reduce code duplication (and also allows the tower to prioritize the repair targets)
- I made the tower energy level requirements stricter (50% -> 80%)
- I removed the range restriction from the repair target search. I think it's good that the towers repair distant structures, as it would take so long for the builder creeps to do it.

I also noticed a small bug in your code. In the `getRepairTarget()` method in `controller.tower.js`, there's a check in the beginning with this condition: `if (target && target instanceof Structure && target.is != tower.id) { ... }`. The end should probably be `target.id != tower.id`.
